teuthology/orchestra/run.py

```diff
@@ -268,15 +272,39 @@ def copy_to_log(f, logger, loglevel=logging.INFO, capture=None):
     # Work-around for http://tracker.ceph.com/issues/8313
     if isinstance(f, ChannelFile):
         f._flags += ChannelFile.FLAG_BINARY
-
     for line in f:
         if capture:
-            capture.write(line)
+            #out = ensure_str(line)
+            if PY3:
+                if isinstance(line, str):
```

Comment:
instead, i'd suggest
```py
if isinstance(capture, io.StringIO):
  capture.write(six.ensure_binary(line))
elif isinstance(capture, io.BytesIO):
  capture.write(six.ensure_str(line, encoding='utf-8', errors='replace')
else:
  capture.write(line)
```

this would address the failure of http://pulpito.ceph.com/vasu-2020-03-13_20:11:34-smoke-nautilus-distro-basic-smithi/4854299/